tls/s2n_connection.c

```diff
@@ -1348,10 +1348,15 @@ int s2n_connection_get_peer_cert_chain(const struct s2n_connection *conn, struct
 {
     POSIX_ENSURE_REF(conn);
     POSIX_ENSURE_REF(cert_chain_and_key);
+    POSIX_ENSURE_REF(cert_chain_and_key->cert_chain);
+
+    /* Ensure that cert_chain_and_key is empty BEFORE we modify it in any way.
+     * That includes before tying its cert_chain to DEFER_CLEANUP.
```

Comment:
Since DEFER_CLEANUP is a s2n specific test method can we also add this check in that macro instead?

Reply:
I'm not sure I follow. Are you suggesting that DEFER_CLEANUP shouldn't be allowed on anything that isn't NULL? That's usually not the case. Think about `DEFER_CLEANUP(s2n_connection *conn = s2n_connection_new(S2N_CLIENT)` or `DEFER_CLEANUP(struct s2n_blob blob = { 0 }, s2n_free)`. The issue here isn't that the thing being DEFER_CLEANUPed isn't NULL, it's just that in this case it could be something we don't actually want to clean up.

Reply:
I was more curious if this check will be able to catch if "cert_chain [is tried] to DEFER_CLEANUP". I think it wont.. so was wondering if there is better place to comment/a way to enforce it.

Reply:
No. This isn't common behavior, this is behavior specific to this use case and this null check. We want cert_chain tied to DEFER_CLEANUP, just only after we've ensure that it's our new cert_chain and not some random old one accidentally passed in.